test-framework/junit5-internal/src/main/java/io/quarkus/test/QuarkusUnitTest.java

```diff
@@ -489,16 +495,23 @@ public boolean test(String s) {
                 builder.addClassLoaderEventListeners(this.classLoadListeners);
                 curatedApplication = builder.build().bootstrap();
 
-                runningQuarkusApplication = new AugmentActionImpl(curatedApplication, customizers, classLoadListeners)
-                        .createInitialRuntimeApplication()
-                        .run(commandLineParameters);
-                //we restore the CL at the end of the test
-                Thread.currentThread().setContextClassLoader(runningQuarkusApplication.getClassLoader());
+                StartupAction startupAction = new AugmentActionImpl(curatedApplication, customizers, classLoadListeners)
+                        .createInitialRuntimeApplication();
+
+                // we restore the CL at the end of the test
+                Thread.currentThread().setContextClassLoader(startupAction.getClassLoader());
```

Comment:
@stuartwdouglas I would expecially would like to be sure I didn't break anything using the `StartupAction` CL. I need a CL before starting the application so that's why I used this one and AFAICS, it's OK as it's the same that is used in the running application.

Reply:
This should be fine